### DISCUSSION TITLE
[Run issue]: mpirun -np Nprocessors to run my input?

### DISCUSSION THREAD

**Discussion by TCFF01** [OpenTOPAS/OpenTOPAS#24] [Feature-Request] [answered]
### Feature request

Hey guys 
can use mpirun -np Nprocessors to run my input?
I'm trying to use it but it doesn't work.
thanks
Telma

### TOPAS category

Outcome model

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by jschuemann** [comment]
Hi Telma,

could you explain a little bit more what you are exactly trying to do?
Thanks.

**Reply by TCFF01** [reply]
I have got a cluster Dell T560 workstation with 48 CPUs and 148 GB RAM.

I installed TOPAS and TOPASnBio but I can't run it using all my processors or part of it. 

I have tried some commands to have it running faster:
first command using 1 processor:  ~/topas/bin/topas teste-phantom.txt then it used one processor.
When I try mpiexec -n 30 ~/topas/bin/topas teste-phantom.txt  - it run 30 times... each one using one processor
I tried  mpirun -np 30 ~/topas/bin/topas teste-phantom.txt  - it run 30 times... each one using one processor

Maybe I should try something different?

**Reply by TCFF01** [reply]
I tried to use the command i:Ts/NumberOfThreads = N  parameter in the input file  but it doesn't work for me. 
I mean, it does not use more than 1 CPU.

**Comment by icmeyer** [comment]
Hi Telma,

Currently multithreading is supported using the "i:Ts/NumberOfThreads = N" parameter in the input file. There is no mpi support for OpenTOPAS. If you are running on a computer cluster and want to take advantage of multiple nodes at once, the approach that has been used with TOPAS before is to create multiple copies of the same job in many directories and then collate the resulting tallies after all the jobs have finished.

**Reply by TCFF01** [reply]
I tried to use the command i:Ts/NumberOfThreads = N  parameter in the input file  but it doesn't work for me. 
I mean, it does not use more than 1 CPU.

**Comment by TCFF01** [comment]
Hey Jan,

I have got a cluster Dell T560 workstation with 48 CPUs and 148 GB RAM.

I installed TOPAS and TOPASnBio but I can't run it using all my processors
or part of it.

I have tried some commands to have it running faster:
first command using 1 processor:  ~/topas/bin/topas teste-phantom.txt then
it used one processor.
When I try mpiexec -n 30 ~/topas/bin/topas teste-phantom.txt  - it run 30
times... each one using one processor
I tried  mpirun -np 30 ~/topas/bin/topas teste-phantom.txt  - it run 30
times... each one using one processor

Maybe I should try something different?



-- 

Profa. Telma C. F. Fonseca
Escola de Engenharia
Universidade Federal de Minas Gerais (UFMG)
Av. Antônio Carlos, 6627, Campus Pampulha
Belo Horizonte, MG, Brasil, 31270-010
Phone: (Office) +55 031 3409-4857
Mobile phone +55 031 9539-4463
e-mail: ***@***.***
e-mail: ***@***.*** / ***@***.***
Site: http://telmafonseca.com.br/

**Reply by JoseRamosMendez** [reply, marked as answer]
I think it is system-related. I guess you're using an executable built in another system in your current Dell system. I would build both Geant4 and OpenTopas in your Dell system from scratch as shown here https://github.com/OpenTOPAS/OpenTOPAS/blob/master/OpenTOPAS_quickStart_Debian.md

**Reply by TCFF01** [reply]
perfect! I will try it again... let you know.
Many thanks Jose :D

**Comment by TCFF01** [comment]
Thanks for the tip!

-- 

Profa. Telma C. F. Fonseca
Escola de Engenharia
Universidade Federal de Minas Gerais (UFMG)
Av. Antônio Carlos, 6627, Campus Pampulha
Belo Horizonte, MG, Brasil, 31270-010
Phone: (Office) +55 031 3409-4857
Mobile phone +55 031 9539-4463
e-mail: ***@***.***
e-mail: ***@***.*** / ***@***.***
Site: http://telmafonseca.com.br/

**Comment by TCFF01** [comment]
Thank you for your feedback and sorry for my delayed response. I have been very busy these last few days and couldn't continue looking into the issue. I will do so and provide you with feedback. Thanks again.
